packages/mui-utils/src/useLocalStorageState/useLocalStorageState.ts

```diff
@@ -150,6 +149,6 @@ function useLocalStorageStateBrowser(
   return [storedValue, setStoredValue];
 }
 
-export default typeof window === 'undefined'
+export default typeof window === 'undefined' || !window.localStorage
```

Comment:
I doubt this is enough. AFAIK, the browser keps it defined bug throw an error when trying to access it https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#feature-detecting_localstorage

Reply:
https://github.com/mui/mui-x/blob/8e90e7de79daff93ba3fca09fe5593573e0d22b9/packages/x-data-grid/src/utils/utils.ts#L13

Reply:
Thanks, I'm adding the try/catch and aligning it with the disabled case (key === null)